src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/resendtogaps/ResendToGapsMessageHandler.java

```diff
@@ -0,0 +1,35 @@
+package uk.gov.hmcts.reform.sscs.ccd.presubmit.resendtogaps;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.sscs.ccd.domain.HearingRoute;
+import uk.gov.hmcts.reform.sscs.ccd.domain.HearingState;
+import uk.gov.hmcts.reform.sscs.model.hearings.HearingRequest;
+import uk.gov.hmcts.reform.sscs.service.servicebus.HearingMessagingServiceFactory;
+
+@Service
+public class ResendToGapsMessageHandler {
+
+    private final boolean gapsSwitchOverFeatureEnabled;
+    private final HearingMessagingServiceFactory hearingMessagingServiceFactory;
+
+    public ResendToGapsMessageHandler(@Value("${feature.gaps-switchover.enabled}") boolean gapsSwitchOverFeatureEnabled,
+                                      @Autowired HearingMessagingServiceFactory hearingMessagingServiceFactory) {
+        this.gapsSwitchOverFeatureEnabled = gapsSwitchOverFeatureEnabled;
+        this.hearingMessagingServiceFactory = hearingMessagingServiceFactory;
+    }
+
+    public void sendMessage(final String ccdCaseId,
+                            final HearingRoute hearingRoute,
+                            final HearingState hearingState) {
+        if (gapsSwitchOverFeatureEnabled) {
```

Comment:
It should only send the message if the gaps switchover feature is enabled - otherwise it won't do anything. If it's not clear I can move the flag up to the handler if that makes more sense?

Reply:
It should send the message if the flag is enabled and the case had a hearing route of GAPS

Reply:
How do we determine this? Would we not assume that if this handler is triggered that the hearing route was GAPS or am I wrong here?